src/components/connection_handler/src/connection_handler_impl.cc

```diff
@@ -461,16 +461,22 @@ void ConnectionHandlerImpl::OnSessionStartedCallback(
     return;
   }
 #endif  // ENABLE_SECURITY
-  sync_primitives::AutoReadLock lock(connection_list_lock_);
-  ConnectionList::iterator it =
-      connection_list_.find(primary_connection_handle);
-  if (connection_list_.end() == it) {
+
+  Connection* connection = nullptr;
+  {
```

Comment:
propose to wrap this if with some kind of lambda like `find connection` and use lock inside